go/resource-overrides/resource-overrides.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/aws/aws-cdk-go/awscdk"
+	"github.com/aws/aws-cdk-go/awscdk/awss3"
+	"github.com/aws/constructs-go/constructs/v3"
+	"github.com/aws/jsii-runtime-go"
+)
+
+type ResourceOverridesStackProps struct {
+	awscdk.StackProps
+}
+
+func NewResourceOverridesStack(scope constructs.Construct, id string, props *ResourceOverridesStackProps) awscdk.Stack {
+	var sprops awscdk.StackProps
+	if props != nil {
+		sprops = props.StackProps
+	}
+	stack := awscdk.NewStack(scope, &id, &sprops)
+
+	// The code that defines your stack goes here
+
+	// otherBucket := awss3.NewBucket(stack, jsii.String("Other"), nil);
+
+	bucket := awss3.NewBucket(stack, jsii.String("MyBucket"), &awss3.BucketProps{
+			Versioned: jsii.Bool(true),
+			Encryption: awss3.BucketEncryption_KMS_MANAGED,
+	});
+
+	defaultChild := bucket.Node().DefaultChild()
+	bucketResource2, ok := defaultChild.(awss3.CfnBucket)
+
+	if !ok {
```

Comment:
The casting here fails. I wonder if i'm missing something and there is another way to do it.

Possibly related to https://github.com/aws/jsii/blob/main/packages/@jsii/go-runtime-test/project/compliance_test.go#L749

?

Reply:
cc @MrArnoldPalmer @RomainMuller @eladb

Reply:
What is the type you are getting?